From: Jinoh Kang <jinoh.kang.kr@gmail.com>
Subject: [PATCH v4 08/13] loader: Fix return type of get_auxiliary().
Message-Id: <6ab88db5-2f5b-e055-e908-9b60f18cc516@gmail.com>
Date: Sat, 29 Jan 2022 03:40:43 +0900
In-Reply-To: <b9568d9a-6539-b275-6bb8-f1441186d0d0@gmail.com>
References: <b9568d9a-6539-b275-6bb8-f1441186d0d0@gmail.com>

This is required for fetching pointer-valued vectors (e.g.
AT_SYSINFO_EHDR).

Signed-off-by: Jinoh Kang <jinoh.kang.kr@gmail.com>
---
 loader/preloader.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/loader/preloader.c b/loader/preloader.c
index f0c0f8f58fd..4a49f6e27a3 100644
--- a/loader/preloader.c
+++ b/loader/preloader.c
@@ -1103,7 +1103,7 @@ static void set_auxiliary_values( struct preloader_state *state,
  *
  * Get a field of the auxiliary structure
  */
-static int get_auxiliary( struct wld_auxv *av, int type, int def_val )
+static ElfW(Addr) get_auxiliary( struct wld_auxv *av, int type, ElfW(Addr) def_val )
 {
   for ( ; av->a_type != AT_NULL; av++)
       if( av->a_type == type ) return av->a_un.a_val;

-- 
2.35.1

